packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts

```diff
@@ -30,10 +32,16 @@ const waitForSignal = async (...signallers: { once(e: "signal", l: () => void):
 		),
 	);
 
+type RuntimeType = IFluidDataStoreRuntime | ContainerRuntime;
+
 describeCompat("TestSignals", "FullCompat", (getTestObjectProvider) => {
 	let provider: ITestObjectProvider;
 	let dataObject1: ITestFluidObject;
 	let dataObject2: ITestFluidObject;
+	let dataObject3: ITestFluidObject;
+	let user1ContainerRuntime: ContainerRuntime;
+	let user2ContainerRuntime: ContainerRuntime;
+	let user3ContainerRuntime: ContainerRuntime;
 
```

Comment:
The less state that exists in blocks common to every test the better. Even more true when some tests don't use the state.
There isn't a lot of setup for these; so they don't need extracted as common.

My earlier comment was about the test patterns that are so similar rather than the setup. I'll try to post a draft of what I am referring to with the existing code.

Reply:
Looks like you followed the common part I was thinking of with the new tests. :) So no example from me.